qa/tasks/salt_manager.py

```diff
@@ -25,7 +25,7 @@
 from util import get_remote_for_role
 
 log = logging.getLogger(__name__)
-master_role = 'client.salt_master'
+master_role = 'salt.master'
```

Comment:
Instead of this, it would be better to have the *role* parameter to SaltManager constructor as an optional argument, like:
```
  def __init__(self, ctx, role=master_role):
        """
        :param ctx: give a hint what is ctx
        :param role: give a hint what is the role
        """
        self.ctx = ctx
        self.master_remote = get_remote_for_role(self.ctx, role)
```
Give the descriptions what `ctx` is, and that `role` is the master role, and how is it used, if there is a section describing what this master role means, then give a ref to where reader can find it.

And later in ceph-salt, use constructor like:
```
ceph_salt_ctx['salt_manager_instance'] = SaltManager(self.ctx, 'salt.master')
```
Split this change off the commit, so we can backport it to SES6 and SES5 if required.

Reply:
@smithfarm if you're okay with those master_role value changed by @gekios I don't insist on my change request.

Reply:
I'm fine with the `client.salt_master` -> `salt.master` change and I'd rather not refactor SES5 and SES6 if it can be avoided.

Reply:
hmm I realized that I had refactored the cephadm role change so it should work without changing anything on salt_manager... I will add the suggested change in saltmanager constructor cause it can be useful.

Reply:
> I'm fine with the `client.salt_master` -> `salt.master` change and I'd rather not refactor SES5 and SES6 if it can be avoided.

this refactor is non-destructive, and does not change the behavior of SaltMaster for ses5 and ses6

as I said, since @smithfarm is okay, I'm okay